### ISSUE TITLE
Detailed training process 

### ISSUE THREAD

**Issue by Hyeokreal** [akurgat/automating-technical-analysis#8]
Hi, Thanks for sharing this great project !

You've provided the model file but I cannot find the details for training the model.
Can I know more details on the training process ?

Thanks.

**Comment by akurgat** [comment]
Hello Hyeokryeol, 

Sure. I am more than willing to share more details.

Both models were built from the following Transformer encoder structure: (line 7 and 8)
https://github.com/jeffheaton/t81_558_deep_learning/blob/master/t81_558_class_10_5_keras_transformers.ipynb

A few adjustments were made to the structure, including: 
1.	Addition of an Average pooling 1D layer preceding the Global Average Pooling layer, using: 
a.	Data format: channel first
b.	Pool size: 1
2.	A legacy Adam optimizer was used to compile the model using:
a.	Learning rate: 1e-4
b.	Decay: 1e-6
c.	Loss: categorical crossentropy (for the action prediction model). 

The model building was done using the following parameters:
1.	Head size: 64
2.	Number of heads: 2
3.	Forward feeding dimensions: 64
4.	Number of transformer blocks: 4
5.	Multilayer perception units: 256
6.	Multilayer perception dropout: 0.3
7.	Dropout: 0.25

The training process was conducted using a combination of differnet stocks and cryptocurrency data, captured in different intervals, ranging from 1 minute to 1 week time horizons, resulting to 135 unique training datasets and another 26 testing datasets used on both models. 

The training was done using the following parameters:
1.	Batch size: 128
2.	Timesteps: 60
3.	Epochs: 15
4.	Callbacks: Earlystopping with:
a.	Patience: 10
b.	Monitor: val_loss
c.	Restore best weights: True

I hope the above comprehensively provides you with the missing details. In case of anything feel free to reach out.

Kind regards,

Andrew Kurgat.

**Comment by mrzhjyw** [comment]
Can you share your thoughts on how to deal with different stock datasets? @akurgat
